### PR TITLE
Bluetooth: Controller: Enqueue missing ISO PDUs with invalid status

### PR DESCRIPTION
**Pull request by cvinayak**
Add implementation to generate node rx for missing ISO PDUs
and set the status as invalid. This is required for ISOAL to
correctly track the sequence numbers for every SDU interval.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c [hunk 1]
```diff
@@ -597,6 +585,15 @@ static void isr_rx(void *param)
...
 				iso_rx_put(node_rx->hdr.link, node_rx);
+			} else {
+				node_rx = ull_iso_pdu_rx_alloc_peek(3U);
+				if (node_rx) {
+					ull_iso_pdu_rx_alloc();
+					isr_rx_iso_data_invalid(lll, bis_idx,
+								bn, node_rx);
+
+					iso_rx_put(node_rx->hdr.link, node_rx);
+				}
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c:596]: Can you elaborate on why we need to peek 3 RX here? And preferably also add that as a comment to the code?

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c [hunk 2]
```diff
@@ -776,6 +773,39 @@ static void isr_rx(void *param)
...
+	node_rx->hdr.type = NODE_RX_TYPE_ISO_PDU;
+	node_rx->hdr.handle = lll->stream_handle[bis_idx];
+
+	iso_meta = &node_rx->hdr.rx_iso_meta;
+	iso_meta->payload_number = lll->payload_count + (lll->bn_curr - 1U) +
+				   (lll->ptc_curr * lll->pto) - lll->bn;
+	iso_meta->timestamp = HAL_TICKER_TICKS_TO_US(radio_tmr_start_get()) +
+			      radio_tmr_aa_restore() - addr_us_get(lll->phy) +
+			      (ceiling_fraction(lll->ptc_curr, lll->bn) *
+			       lll->iso_interval * PERIODIC_INT_UNIT_US);
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c:790]: Are these calculations from the core spec? If so, please add a reference to the section, and if not please ignore this comment :)

**Review by Thalley — CHANGES_REQUESTED** [review]
LGTM but would like additional documentation in code

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c (hunk 2 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c:801]: This is not from spec, but vendor specific capture of the anchor point for a ISO interval.

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c]: Added comments in code.

**Review by Thalley — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Pull request merged**